src/Microsoft.VisualStudio.Composition/ExportProvider.cs

```diff
@@ -989,6 +1097,16 @@ protected internal abstract class PartLifecycleTracker : IDisposable
             /// </remarks>
             private HashSet<PartLifecycleTracker> deferredInitializationParts;
 
+            /// <summary>
+            /// A lazily-initialized collection of all transitive non-shared parts that should be disposed with this non-shared part.
+            /// </summary>
+            private HashSet<PartLifecycleTracker> nonSharedChildParts;
```

Comment:
Thank you for walking me through the changes 

Assuming I understood correctly that a DAG of non-shared parts get flattened into a single level, is it possible (and I'm not sure it is) for me as a consumer to want to release an export that was maybe 3 levels deep and expect only its children to be released as well (but instead the entire group (siblings, ancestors, etc) gets released)? 

I _think_ the answer is no because I would likely be importing and holding onto the root and wouldn't release any transitive dependencies - just release the root when I'm finished. 

But.... Just wanted to check anyway.

Reply:
Good question, and correct deduction. The only `Export` or `Lazy<T>` exposed to the caller is the "root" non-shared part. So the only thing they could release is the root level one.